src/tests/BuildWasmApps/Wasm.Build.Tests/ICUShardingTests.cs

```diff
@@ -0,0 +1,93 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using Xunit;
+using Xunit.Abstractions;
+using System.Globalization;
+
+#nullable enable
+
+namespace Wasm.Build.Tests
+{
+    public class ICUShardingTests : BuildTestBase
+    {
+        public ICUShardingTests(ITestOutputHelper output, SharedBuildPerTestClassFixture buildContext)
+            : base(output, buildContext)
+        {
+        }
+
+        public static IEnumerable<object?[]> ICUShardingTestData(bool enableSharding, RunHost host)
+            => ConfigWithAOTData(false)
+                .Multiply(
+                    new object?[] { enableSharding, "en", "en-GB", false},      //EFIGS
+                    new object?[] { enableSharding, "en", "zh", true},          //EFIGS
+                    new object?[] { enableSharding, "es-ES", "am-ET", true},    //EFIGS
+                    new object?[] { enableSharding, "es", "fr-FR", true},       //EFIGS
+                    new object?[] { enableSharding, "zh", "zh-Hans", false},    //CJK
+                    new object?[] { enableSharding, "ko", "zh-Hans", false},    //CJK
+                    new object?[] { enableSharding, "ja-JP", "es-ES", true},    //CJK //this one should have thrown
+                    new object?[] { enableSharding, "am-ET", "de-DE", false},   //no_CJK
+                    new object?[] { enableSharding, "am-ET", "ko", true},       //no_CJK
+                    new object?[] { enableSharding, "am-ET", "ja-JP", true},    //no_CJK
+                    new object?[] { enableSharding, "am-ET", "vi-VN", false},   //no_CJK
+                    new object?[] { enableSharding, null, "zh-Hans", true})     //ALL
```

Comment:
What is being tested here? What do the groupings mean?

Reply:
The groupings are: `<enable sharding switch> <default culture> <test culture> <expect to throw>`. First we set the default culture, which determines which shards to use, and then we try to load the test culture and depending on which default we're using we can expect the test to throw or not.

Reply:
```
                    new object?[] { enableSharding, "en", "en-GB", false},      //EFIGS
                    new object?[] { enableSharding, "en", "zh", true},          //EFIGS
                    new object?[] { enableSharding, "es-ES", "am-ET", true},    //EFIGS
                    new object?[] { enableSharding, "es", "fr-FR", true},       //EFIGS
                    new object?[] { enableSharding, "zh", "zh-Hans", false},    //CJK
                    new object?[] { enableSharding, "ko", "zh-Hans", false},    //CJK
```

Can we split the test data into one per .. shard name? And have multiple `MemberData` attributes on the tests. It should make it clearer if we are missing any cases.

Reply:
Yea that make sense.